source/crud/tests/unified/estimatedDocumentCount.yml

```diff
@@ -0,0 +1,213 @@
+description: "estimatedDocumentCount"
+
+schemaVersion: "1.0"
+
+createEntities:
+  - client:
+      id: &client0 client0
+      uriOptions: { retryReads: false } # needed to set failPoint
```

Comment:
Needed to set `retryReads` to `false` on the client so that the failpoints I set for `aggregate` and `count` aren't retried (as they would be by default). I could alternatively set a failpoint with `times: 2` and expect two attempts, but this seems easier.

Reply:
Were the network error tests particularly relevant? Reading https://github.com/mongodb/specifications/pull/905#discussion_r572211505, it looks like @jyemin was more concerned with testing a _server error_ other than NamespaceNotFound(26).

That said, I don't mind if we leave the network error tests in place, since it will give us an opportunity to use the `isClientError: true` assertion.

I would suggest revising the comment to be more clear/accurate, since this isn't relevant to all `failPoint` operations. For example:

```
# Avoid retrying fail points with closeConnection
```

Reply:
We discussed offline; they seemed important to keep. Java and Go have three potential error paths for the new aggregate estimatedDocumentCount: 

1. `NamespaceNotFound` error (tested by nonexistent collection)
2. A command error that is not `NamespaceNotFound` (tested by the new failpoints w/ error codes)
3. All other errors (tested by this network failure)

I've revised the comment above, but the `isClientError: true` assertion fails. Will elaborate in the other thread about `isClientError`.